pkg/controller/installation/bootstrapReconciler.go

```diff
@@ -158,6 +165,79 @@ func getConsoleRouteCR(ctx context.Context, serverClient k8sclient.Client) (*rou
 	return consoleRouteCR, nil
 }
 
+func (r *Reconciler) reconcilerGithubOauthSecret(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+
+	githubOauthSecretCR := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.ConfigManager.GetGHOauthClientsSecretName(),
+			Namespace: r.ConfigManager.GetOperatorNamespace(),
+		},
+	}
+
+	err := serverClient.Get(ctx, k8sclient.ObjectKey{Name: githubOauthSecretCR.Name, Namespace: githubOauthSecretCR.Namespace}, githubOauthSecretCR)
+	if !k8serr.IsNotFound(err) && err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	} else if k8serr.IsNotFound(err) {
+		githubOauthSecretCR.Data = map[string][]byte{
+			"clientId": []byte("dummy"),
+			"secret":   []byte("dummy"),
+		}
+	}
+	githubOauthSecretCR.ObjectMeta.ResourceVersion = ""
+
+	err = resources.CreateOrUpdate(ctx, serverClient, githubOauthSecretCR)
```

Comment:
I think a `serverClient.Create` in an `else if` branch above, would work for us.

btw: I think we will eventually remove our `resources.CreateOrUpdate` function once we switch to `controllerutil.CreateOrUpdate` in our code.

Reply:
Or as an alternative to a `serverClient.Get` + conditions, you can do a `serverClient.Create` and ignore "AllreadyExists" errors. :)